src/MainWindow.vala

```diff
@@ -817,4 +840,149 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
+//        if (entry.get_text () != "") {
+//            try {
+//                var output = eval.evaluate (text, decimal_places);
+//                if (entry.get_text () != output) {
+//                    entry.set_text (output);
+//                }
+//            } catch (Core.OUT_ERROR e) {
+//                    infobar_label.label = e.message;
+//                    infobar.revealed = true;
+//            }
+//        }
+//    }
+
+    private bool on_key_press (Gtk.EventControllerKey controller, uint keyval, uint keycode, Gdk.ModifierType mod_state) {
+        switch (keyval) {
+            case Gdk.Key.@0:
+            case Gdk.Key.KP_0:
+                button_0.activate();
+                return true;
+            case Gdk.Key.@1:
+            case Gdk.Key.KP_1:
+                button_1.activate();
+                return true;
+            case Gdk.Key.@2:
+            case Gdk.Key.KP_2:
+                button_2.activate();
+                return true;
+            case Gdk.Key.@3:
+            case Gdk.Key.KP_3:
+                button_3.activate();
+                return true;
+            case Gdk.Key.@4:
+            case Gdk.Key.KP_4:
+                button_4.activate();
+                return true;
+            case Gdk.Key.@5:
+            case Gdk.Key.KP_5:
+                button_5.activate();
+                return true;
+            case Gdk.Key.@6:
+            case Gdk.Key.KP_6:
+                button_6.activate();
+                return true;
+            case Gdk.Key.@7:
+            case Gdk.Key.KP_7:
+                button_7.activate();
+                return true;
+            case Gdk.Key.@8:
+            case Gdk.Key.KP_8:
+                button_8.activate();
+                return true;
+            case Gdk.Key.@9:
+            case Gdk.Key.KP_9:
+                button_9.activate();
+                return true;
+            case Gdk.Key.plus:
+            case Gdk.Key.KP_Add:
+                button_add.activate();
+                return true;
+            case Gdk.Key.minus:
+            case Gdk.Key.KP_Subtract:
+                button_sub.activate();
+                return true;
+            case Gdk.Key.asterisk:
+            case Gdk.Key.KP_Multiply:
+                button_mult.activate();
+                return true;
+            case Gdk.Key.slash:
+            case Gdk.Key.KP_Divide:
+                button_div.activate();
+                return true;
+            case Gdk.Key.period:
+            case Gdk.Key.decimalpoint:
+            case Gdk.Key.KP_Decimal:
+                button_point.activate();
+                return true;
+            case Gdk.Key.BackSpace:
+            case Gdk.Key.KP_Delete:
+                button_del.activate();
+                return true;
+            case Gdk.Key.Return:
+            case Gdk.Key.KP_Enter:
+            case Gdk.Key.KP_Equal:
```

Comment:
```suggestion
            case Gdk.Key.KP_Equal:
            case Gdk.Key.equal:
```

Reply:
case Gdk.Key.Return: is needed as in my code the display is in read only mode